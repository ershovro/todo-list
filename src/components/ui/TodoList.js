import React from 'react';
import PropTypes from 'prop-types';

const resolveClassName = ({isCompleted}) => (
   `todoList__item ${isCompleted ? 'todoList__item_completed' : ''}`
);
const TodoList = ({todos = [], toggleCompletedState = f => f}) => (
   <div className="todoList">
      {
         !todos.length
            ? <p>No Todos. (Add a todo)</p>
            : <ul>
               { todos.map( todo =>
                  <li
                     key={todo.id}
                     className={ resolveClassName(todo) }
                     onClick={ () => toggleCompletedState(todo.id) }>
                     {todo.text}
                  </li>
               ) }
              </ul>
      }
   </div>
);

TodoList.propTypes = {
   todos: PropTypes.array,
   toggleCompletedState: PropTypes.func
};

export default TodoList;