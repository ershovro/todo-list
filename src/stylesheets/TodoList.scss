.todoList__item_completed {
  text-decoration: line-through;
}