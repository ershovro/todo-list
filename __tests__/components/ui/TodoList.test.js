import React from 'react';
import TodoList from "../../../src/components/ui/TodoList";
import enzyme from 'enzyme';
import { JSDOM } from 'jsdom';
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
import Adapter from 'enzyme-adapter-react-16';


const { window } = new JSDOM(`...`);
const { document } = (new JSDOM(`...`)).window;

enzyme.configure({ adapter: new Adapter() });
global.window = window;
global.document = document;


describe('<TodoList /> ui component', () => {

   describe('Rendering ui', () => {

      it('Renders Correctly', () => {
         compose(expect, toJSON, enzyme.shallow)(
            <TodoList todos={_testTodos} />
         ).toMatchSnapshot();
      });

      it('Renders div with class .todoList', () => {
         expect(
            enzyme.mount( <TodoList /> ).
            find('div.todoList').
            length
         ).toBe(1)
      });

      it('rendering without to-do list', () => {
         expect(
            enzyme.mount( <TodoList /> ).
            find('p').text()
         ).toBe('No Todos. (Add a todo)')
      });

      it('clicking on an todo does not cause an error', () => {
         const wrapper = enzyme.mount(  <TodoList todos={_testTodos}/> );

         wrapper.find('.todoList__item').first().simulate('click');
      });

   });

});
