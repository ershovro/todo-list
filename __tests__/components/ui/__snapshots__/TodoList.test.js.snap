// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> ui component Rendering ui Renders Correctly 1`] = `
<div
  className="todoList"
>
  <ul>
    <li
      className="todoList__item todoList__item_completed"
      key="0"
      onClick={[Function]}
    />
    <li
      className="todoList__item todoList__item_completed"
      key="1"
      onClick={[Function]}
    />
    <li
      className="todoList__item "
      key="2"
      onClick={[Function]}
    />
  </ul>
</div>
`;
